tools/osbuilder/rootfs-builder/rootfs.sh

```diff
@@ -42,6 +40,12 @@ trap 'handle_error $LINENO' ERR
 # Default architecture
 ARCH=$(uname -m)
 
+# Default Rust & MUSL versions
+get_kata_rust_version ||
+    die "Could not detect the required rust version for AGENT_VERSION='${AGENT_VERSION:-main}'."
+get_kata_musl_version ||
+    die "Could not detect the required musl version for AGENT_VERSION='${AGENT_VERSION:-main}'."
```

Comment:
Nit: How about setting `AGENT_VERSION` to simplify this and avoid duplicated defaults?

```suggestion
AGENT_VERSION=${AGENT_VERSION:-main}

get_kata_rust_version ||
    die "Could not detect the required rust version for AGENT_VERSION='${AGENT_VERSION}'."
get_kata_musl_version ||
    die "Could not detect the required musl version for AGENT_VERSION='${AGENT_VERSION}'."
```

Same comment for the calls in `test_images.sh`.

Reply:
And in the lines 47 and 49: `s/detect/get/`

Reply:
> Nit: How about setting `AGENT_VERSION` to simplify this and avoid duplicated defaults?

There are other places which use `AGENT_VERSION` and expect the default value to be empty.  Disentangling that isn't really in scope for this patchset.

Reply:
> And in the lines 47 and 49: `s/detect/get/`

I've made this change.